### PR TITLE
GC: Precisely explain what objects are retained

### PR DESCRIPTION
**Pull request by arielshaqed**

### CODE REVIEW

File: docs/reference/garbage-collection.md [hunk 1]
```diff
@@ -36,17 +36,80 @@ That is, it is hard-deleted only after the retention period has ended for all re
...
+
+Garbage collection is configured by specifying the number of days to retain
+objects on each branch.  If a branch is configured to retain objects for a
+given number of days, any object that was accessible from the HEAD of a
+branch in that past number of days will be retained.
+
+The garbage collection process proceeds in two main phases:
+
+* **Discover which commits will retain their objects.**  For every branch,
+  GC looks at the HEAD of the branch that many days ago; every commit at or
```
Comment (Jonathan-Rosenberg) [inline docs/reference/garbage-collection.md:63]:
```suggestion
  the garbage collector looks at the HEAD of the branch that many days ago; every commit at or
```

File: docs/reference/garbage-collection.md [hunk 2]
```diff
@@ -36,17 +36,80 @@ That is, it is hard-deleted only after the retention period has ended for all re
...
+    `2022-03-09`.  So that commit is retained (along with all more recent
+    commits on `main`), but commits `2022-02-27` and `2022-03-01` will be
+    collected.
+
+* **Discover which objects need to be garbage collected.** Hold (_only_)
+  objects accessible on some retained commits.
+
+  In the example, all objects of commit `2022-03-12`, for instance, are
+  retained.  This _includes_ objects added in previous commits.  However,
+  objects added in commit `d: 2022-03-14` and overwritten in commit `d:
```
Comment (Jonathan-Rosenberg) [inline docs/reference/garbage-collection.md:104]:
```suggestion
  objects added in commit `d: 2022-03-14` and overwritten or deleted in commit `d:
```
?

**Review by Jonathan-Rosenberg — APPROVED** [review]
LGTM
Added a few comments of some stuff that seemed not that clear to me

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Comment by arielshaqed** [comment]
Pulling untested by adminiforce as it's just docs.

**Pull request merged**